### PR TITLE
feat(integ): add Deadline secrets management integration tests

### PR DESCRIPTION
**Pull request by jericht**
Adds a new suite of integration tests for Secrets Management that test:
1. The `Repository` construct runs the installer with the correct SM configuration
2. The `RenderQueue` construct configures itself as the Server role with SM'

Other minor changes include:
- Updates the `RenderStruct` to use HTTPS as the internal protocol when created in HTTPS mode (required by SM)
- Renamed `ThinkboxDockerImageOverrides.ts` to `thinkbox-docker-image-overrides.ts` to conform to the rest of the filenames.

### Testing
- Ran the integration tests locally (with Docker recipes + Deadline installer that had the SM changes) and verified they passed.
- Deployed a modified version of our AutoBump project to run these tests in parallel and verified it passed.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: integ/components/deadline/deadline_05_secretsManagement/lib/secretsManagement-testing-tier.ts
Comment (jusiskin) [inline integ/components/deadline/deadline_05_secretsManagement/lib/secretsManagement-testing-tier.ts]: will these two calls to `this.configureCert(testSuiteId, ...)` cause conflicting construct IDs in the scope?

File: integ/components/deadline/deadline_05_secretsManagement/lib/secretsManagement-testing-tier.ts
Comment (jusiskin) [inline integ/components/deadline/deadline_05_secretsManagement/lib/secretsManagement-testing-tier.ts]: Looks like the method documentation might have been copy/pasted and not updated (re: "output the name of its log group"). Do you mind updating it?

Also, I wonder if we can make the method name a little more specific. As it stands, it sounds like we're acting on the repository itself. How about `configureRepoConnection`?

File: integ/components/deadline/deadline_05_secretsManagement/lib/secretsManagement-testing-tier.ts
Comment (jusiskin) [inline integ/components/deadline/deadline_05_secretsManagement/lib/secretsManagement-testing-tier.ts]: This seems like behavior that might benefit any class deriving from the base. Should we hoist this command up into `TestingTier.installDeadlineClient()`?

File: integ/components/deadline/deadline_05_secretsManagement/scripts/bastion/testing/SM-run-secrets-command.sh
Comment (jusiskin) [inline integ/components/deadline/deadline_05_secretsManagement/scripts/bastion/testing/SM-run-secrets-command.sh:35]: I'm sure this is probably necessary, but it is not obvious what the intention of this expression is:

```
${SM_CMD_ARGS[@]+"${SM_CMD_ARGS[@]}"
```

Is there any way we can add a comment to explain it?

File: integ/components/deadline/deadline_05_secretsManagement/test/deadline_05_secretsManagement.test.ts
Comment (jusiskin) [inline integ/components/deadline/deadline_05_secretsManagement/test/deadline_05_secretsManagement.test.ts]: Is it worth being defensive here? After the loop, we could validate that we found the required outputs and failing if not.

**Review by jusiskin — CHANGES_REQUESTED** [review]
Looking good. On a higher level, I wonder if it would make more sense to have these tests as part of the existing repository and render queue suites. Otherwise, just a few small suggestions here.

File: integ/components/deadline/deadline_05_secretsManagement/scripts/bastion/testing/SM-run-secrets-command.sh
Comment (kozlove-aws) [reply integ/components/deadline/deadline_05_secretsManagement/scripts/bastion/testing/SM-run-secrets-command.sh:35]: We already have explanation for such array displaying in `installDeadlinerepository.sh` script. So there is no reason to explain it again in integ tests.

Comment (jusiskin) [reply integ/components/deadline/deadline_05_secretsManagement/scripts/bastion/testing/SM-run-secrets-command.sh:35]: I'm not sure how anyone reading this code would know to look at `installDeadlinerepository.sh` to help understand the code in this file?

File: integ/components/deadline/deadline_05_secretsManagement/test/deadline_05_secretsManagement.test.ts
Comment (jericht) [reply integ/components/deadline/deadline_05_secretsManagement/test/deadline_05_secretsManagement.test.ts]: Good idea, added this in

File: integ/components/deadline/deadline_05_secretsManagement/scripts/bastion/testing/SM-run-secrets-command.sh
Comment (jericht) [reply integ/components/deadline/deadline_05_secretsManagement/scripts/bastion/testing/SM-run-secrets-command.sh:35]: Yeah, I think having another comment like the one we have in the repo installer script is helpful, doesn't hurt to add it in. Updated.

File: integ/components/deadline/deadline_05_secretsManagement/lib/secretsManagement-testing-tier.ts
Comment (jericht) [reply integ/components/deadline/deadline_05_secretsManagement/lib/secretsManagement-testing-tier.ts]: Sure, moved this into the userdata generated in the base class.

File: integ/components/deadline/deadline_05_secretsManagement/lib/secretsManagement-testing-tier.ts
Comment (jericht) [reply integ/components/deadline/deadline_05_secretsManagement/lib/secretsManagement-testing-tier.ts]: Thanks, updated

File: integ/components/deadline/deadline_05_secretsManagement/lib/secretsManagement-testing-tier.ts
Comment (jericht) [reply integ/components/deadline/deadline_05_secretsManagement/lib/secretsManagement-testing-tier.ts]: Good catch, yes it was generating two `CfnOutput` constructs with the same ID. Updated.

**Comment by jericht** [comment]
@jusiskin @kozlove-aws I've address CR feedback, so this is ready for re-review. I'm running all the integration tests as a final e2e, so please hold off on merging until they are finished, even if this gets two approvals.

File: integ/components/deadline/deadline_05_secretsManagement/test/deadline_05_secretsManagement.test.ts
Comment (kozlove-aws) [inline integ/components/deadline/deadline_05_secretsManagement/test/deadline_05_secretsManagement.test.ts]: With only one test case it looks like over complicated. Do you expect another test cases here?

File: integ/lib/testing-tier.ts
Comment (kozlove-aws) [inline integ/lib/testing-tier.ts]: suffix can be `undefined` here.

**Review by kozlove-aws — COMMENTED** [review]
Great work, just a couple tiny moments.

File: integ/components/deadline/deadline_05_secretsManagement/test/deadline_05_secretsManagement.test.ts
Comment (jericht) [reply integ/components/deadline/deadline_05_secretsManagement/test/deadline_05_secretsManagement.test.ts]: Good point. I left this in mostly to conform to the format of the other integ tests we have, but since we don't have any other test cases right now this can be removed. Updated.

File: integ/lib/testing-tier.ts
Comment (jericht) [reply integ/lib/testing-tier.ts]: Fixed

**Comment by jericht** [comment]
@kozlove-aws @jusiskin integ tests have passed, this is ready to merge if you both approve of the changes

**Review by kozlove-aws — APPROVED** [review]

File: integ/components/deadline/deadline_05_secretsManagement/lib/secretsManagement-testing-tier.ts
Comment (jusiskin) [reply integ/components/deadline/deadline_05_secretsManagement/lib/secretsManagement-testing-tier.ts]: Looks good now - thanks!

File: integ/components/deadline/deadline_05_secretsManagement/scripts/bastion/testing/SM-run-secrets-command.sh
Comment (jusiskin) [reply integ/components/deadline/deadline_05_secretsManagement/scripts/bastion/testing/SM-run-secrets-command.sh:35]: Thanks - this is really helpful to readers

File: integ/components/deadline/deadline_05_secretsManagement/test/deadline_05_secretsManagement.test.ts
Comment (jusiskin) [inline integ/components/deadline/deadline_05_secretsManagement/test/deadline_05_secretsManagement.test.ts]: nit: this might help improve the output when this test fails so that we can see the actual deadline command output in the test failure message:

```suggestion
    expect(response.output).toMatch(/ec2-user\s+\[Server\]\s+Registered/);
```

I'm okay deferring this for later since the test will still fail on error and developers can always go to the SSM Run Command console in the account to obtain the actual output.

**Review by jusiskin — APPROVED** [review]
looks great. approved.

I found a minor test forensics/output improvement, but not going to hold up merging this.

File: integ/components/deadline/deadline_05_secretsManagement/test/deadline_05_secretsManagement.test.ts
Comment (jericht) [reply integ/components/deadline/deadline_05_secretsManagement/test/deadline_05_secretsManagement.test.ts]: As discussed offline, I've created a custom jest matcher using `expect.extends` to achieve what we want here with a more helpful error message if the test fails. Let me know what you think

**Review by kozlove-aws — APPROVED** [review]

File: integ/components/deadline/deadline_05_secretsManagement/test/deadline_05_secretsManagement.test.ts
Comment (jusiskin) [reply integ/components/deadline/deadline_05_secretsManagement/test/deadline_05_secretsManagement.test.ts]: Wow - great job! bonus points 🥇 💯

**Review by jusiskin — APPROVED** [review]

**Pull request merged**
